src/docs/DocsWrapper.tsx

```diff
@@ -22,14 +27,30 @@ const paddingIncrement = 2
 
 export default function DocsWrapper({ children }: DocsWrapperProps): ReactElement {
   const theme = useTheme()
-  const { pathname } = useRouter()
+  const router = useRouter()
   const { docsMenuContent, errorMessage } = useContext(DocsMenuContext)
+  const [flattenedPages, setFlattenedPages] = useState<DocsMenuContent>([])
 
   const linkColour = useMemo(
     () => (theme.palette.mode === 'light' ? theme.palette.primary.main : theme.palette.secondary.main),
     [theme]
   )
 
+  const flattenPages = useCallback((array) => {
+    let result: DocsMenuContent = []
+    array.forEach((a) => {
+      result.push(a)
+      if (Array.isArray(a.children)) {
+        result = result.concat(flattenPages(a.children))
+      }
+    })
+    return result.filter((item: any) => item.hasIndex || item.hasIndex === undefined)
+  }, [])
+
+  useEffect(() => {
```

Comment:
You could condense this effect hook and the above state hook into a single memo hook:
```
  const flattenedPages = useMemo(() => flattenPages(docsMenuContent), [docsMenuContent, flattenPages])
```